test/extension-tests/successful-build/test/prefered-generators.test.ts

```diff
@@ -0,0 +1,213 @@
+import * as chai from 'chai';
+import {expect} from 'chai';
+import * as chaiAsPromised from 'chai-as-promised';
+chai.use(chaiAsPromised);
+
+import {clearExistingKitConfigurationFile} from '../../../test-helpers';
+import {DefaultEnvironment} from '../../../helpers/test/default-environment';
+
+import {CMakeTools} from '../../../../src/cmake-tools';
+import {ITestCallbackContext, IHookCallbackContext} from 'mocha';
+
+interface BuildSystemConfiguration {
+  defaultKit: string;
+  expectedDefaultGenerator: string;
+  path?: string;
+}
+
+let workername = process.env.APPVEYOR_BUILD_WORKER_IMAGE;
+
+if (process.env.TRAVIS_OS_NAME) {
+  workername = process.env.TRAVIS_OS_NAME;
+}
+
+if (workername === undefined) {
+  workername = 'DevPC';
+}
+
+const DefaultCompilerMakeSystem: {[os: string]: BuildSystemConfiguration[]} = {
+  ['DevPC']: [
+    {defaultKit: 'VisualStudio.14.0', expectedDefaultGenerator: 'Visual Studio 14 2015'},
+    {
+      defaultKit: 'GCC',
+      expectedDefaultGenerator: 'MinGW Makefiles',
+      path: 'C:\\Program Files\\mingw-w64\\x86_64-7.2.0-posix-seh-rt_v5-rev1\\mingw64\\bin'
+    },
+    {defaultKit: 'Clang', expectedDefaultGenerator: 'MinGW Makefiles', path: 'C:\\Program Files\\LLVM\\bin'}
+  ],
+  ['Visual Studio 2017']: [
+    {defaultKit: 'Visual Studio Community 2017', expectedDefaultGenerator: 'Visual Studio 15 2017'},
+    {
+      defaultKit: 'GCC',
+      expectedDefaultGenerator: 'MinGW Makefiles',
+      path: 'C:\\mingw-w64\\x86_64-7.2.0-posix-seh-rt_v5-rev1\\mingw64\\bin'
+    }
+  ],
+  ['Visual Studio 2017 Preview']: [
+    {defaultKit: 'Visual Studio Community 2017', expectedDefaultGenerator: 'Visual Studio 15 2017'}
+  ],
+  ['Visual Studio 2015']: [
+    {defaultKit: 'VisualStudio.14.0', expectedDefaultGenerator: 'Visual Studio 14 2015'},
+    {
+      defaultKit: 'GCC',
+      expectedDefaultGenerator: 'MinGW Makefiles',
+      path: 'C:\\mingw-w64\\x86_64-6.3.0-posix-seh-rt_v5-rev1\\mingw64\\bin'
+    }
+  ],
+  ['Visual Studio 2013']: [
+    {defaultKit: 'VisualStudio.11.0', expectedDefaultGenerator: 'Visual Studio 11 2012'},
+    {defaultKit: 'GCC', expectedDefaultGenerator: 'MinGW Makefiles', path: 'C:\\MinGW\\bin'}
+  ],
+  ['linux']: [
+    {defaultKit: 'Clang', expectedDefaultGenerator: 'Unix Makefiles'},
+    {defaultKit: 'GCC', expectedDefaultGenerator: 'Unix Makefiles'}
+  ],
+  ['osx']: [{
+    defaultKit: 'Apple',
+    expectedDefaultGenerator: 'Unix Makefiles',
+    path:
+        '/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin;/Applications/Xcode.app/Contents/Developer/usr/bin'
+  }]
+};
```

Comment:
I like the conciseness here.

Reply:
I fixed a problem in the path separation.